db/revtree.go

```diff
@@ -395,3 +405,25 @@ func encodeRevisions(revs []string) Body {
 	}
 	return Body{"start": start, "ids": ids}
 }
+
+// Given a revision history encoded by encodeRevisions() and a list of possible ancestor revIDs,
+// trim the history to stop at the first ancestor revID. If no ancestors are found, trim to
+// length maxUnmatchedLen.
+func trimEncodedRevisionsToAncestor(revs Body, ancestors []string, maxUnmatchedLen int) bool {
+	start, digests := splitRevisionList(revs)
+	if digests == nil {
+		return false
```

Comment:
What's the bool return value signify?  I thought at first it was "nothing trimmed", but I guess it's more like "there's nothing to evaluate for trimming"?  The current caller is ignoring the returned value, and the unit tests don't cover the case where it returns false.